DependencyInjection/MonologExtension.php

```diff
@@ -755,24 +755,35 @@ private function buildHandler(ContainerBuilder $container, $name, array $handler
                         }
                     }
 
-                    $hub = new Definition(
+                    $hubId = \sprintf('monolog.handler.%s.hub', $name);
+                    $hub = $container->setDefinition($hubId, new Definition(
                         'Sentry\\State\\Hub',
                         [new Reference($clientId)]
-                    );
-                    $container->setDefinition(\sprintf('monolog.handler.%s.hub', $name), $hub);
+                    ));
```

Comment:
This never worked correctly. The `monolog.handler.%s.hub` service was registered but never actually used. Instead, the entire definition was passed as an argument, resulting in the creation of two separate services: the unused `monolog.handler.%s.hub` service and an anonymous service that was passed to the handler.